flink-ai-flow/ai_flow/api/ops.py

```diff
@@ -0,0 +1,582 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+from typing import Union, Text, Tuple, Optional, List
+from notification_service.base_notification import UNDEFINED_EVENT_TYPE, ANY_CONDITION
+from ai_flow.workflow.periodic_config import PeriodicConfig
+from ai_flow.client.ai_flow_client import get_ai_flow_client
+from ai_flow.ai_graph.ai_node import AINode
+from ai_flow.util import json_utils
+from ai_flow.graph.channel import Channel, NoneChannel
+from ai_flow.workflow.control_edge import ControlEdge, \
+    TaskAction, EventLife, MetValueCondition, MetCondition, DEFAULT_NAMESPACE, AIFlowInnerEventType
+from ai_flow.ai_graph.ai_graph import default_graph, add_ai_node_to_graph
+from ai_flow.meta.dataset_meta import DatasetMeta
+from ai_flow.meta.model_meta import ModelMeta, ModelVersionMeta
+from ai_flow.context.project_context import project_description
+from ai_flow.context.workflow_context import workflow_config
+
+
+def read_dataset(dataset_info: Union[DatasetMeta, Text, int],
+                 executor=None,
+                 **kwargs) -> Channel:
```

Comment:
It seems that this varargs is not necessary. The parameters here can be categorized to three different types:
1. Parameters to AI Flow - such as dataset_info here, so that AI Flow can help do the bookkeeping like lineage.
2. Parameters to the `Executor` - in this case, the parameters can be passed to the constructor of the `Executor`.
3. Parameters to the job generator - Such parameters passed in here are supposed to be used by the job generator to generate the job for this particular Executor. 

It looks that the varargs here are for the 3rd use case. Arguably, we can also put these per `Executor` parameters in the `Executor` parameters. This essentially lets the `Executor` plugin pass these parameters to the job generator and thus makes the `read_dataset()` API simpler. The same design principle could be applied to the other methods.